implementation/src/main/java/io/smallrye/metrics/MetricsRequestHandler.java

```diff
@@ -144,42 +143,44 @@ public void handleRequest(String requestPath,
     /**
      * Determine which exporter we want.
      *
-     * @param method http method (GET, POST, etc)
+     * @param method        http method (GET, POST, etc)
      * @param acceptHeaders accepted content types
-     *
-     * @return An exporter instance or null in case no matching exporter existed.
+     * @param responder     the responder to use if an error occurs
+     * @return An exporter instance. If an exporter cannot be obtained for some reason, this method will use the responder
+     * to inform the user and will return null.
      */
-    private Exporter obtainExporter(String method, Stream<String> acceptHeaders) {
-        Exporter exporter;
-
+    private Exporter obtainExporter(String method, Stream<String> acceptHeaders, Responder responder) throws IOException {
         if (acceptHeaders == null) {
             if (method.equals("GET")) {
-                exporter = new PrometheusExporter();
+                return new PrometheusExporter();
             } else {
+                responder.respondWith(405, "Prometheus exporter only supports the GET method.", Collections.emptyMap());
                 return null;
             }
         } else {
             // Header can look like "application/json, text/plain, */*"
             if (acceptHeaders.findFirst().map(e -> e.startsWith("application/json")).orElse(false)) {
-
-
                 if (method.equals("GET")) {
-                    exporter = new JsonExporter();
+                    return new JsonExporter();
                 } else if (method.equals("OPTIONS")) {
-                    exporter = new JsonMetadataExporter();
+                    return new JsonMetadataExporter();
                 } else {
+                    responder.respondWith(405, "Only GET and OPTIONS methods are accepted.", Collections.emptyMap());
                     return null;
                 }
             } else {
                 // This is the fallback, but only for GET, as Prometheus does not support OPTIONS
                 if (method.equals("GET")) {
-                    exporter = new PrometheusExporter();
+                    return new PrometheusExporter();
+                } else if (method.equals("OPTIONS")) {
+                    responder.respondWith(406, "OPTIONS method is only allowed with application/json media type.", Collections.emptyMap());
+                    return null;
                 } else {
+                    responder.respondWith(405, "Prometheus exporter only supports the GET method.", Collections.emptyMap());
```

Comment:
Please mention also OPTIONS method, it can be confusing for end user to get different message depending on specified media type. 

This message should be consistent with the one on line 168. I would keep just `Only GET and OPTIONS methods are accepted.`. If user specifies OPTIONS after this message, he will get message about application/json media type requirement. So there is nice path to get the request properly.

Reply:
That's a fair point, will fix that tomorrow

Reply:
> Please mention also OPTIONS method, it can be confusing for end user to get different message depending on specified media type.

Not sure I get this. It is quite common that you can POST a html form, but not a .jpg on a server

Reply:
I think Rostislav's point is that if you use a method that's not valid for any media type at all (PUT for example), the message should be always the same no matter what the media type is. It's basically a generic message saying that you're using it completely wrong. 
I don't really care which approach we pick, it's quite unlikely that anyone will try any other method than GET or OPTIONS, I guess.

Reply:
+1 for consistent message, I prefer simple `Only GET and OPTIONS methods are accepted.`
User can make typo (e.g. just OPTION) or just try what's possible with the endpoint.

Reply:
Got it. Thanks

Reply:
Fixed. Also I refactored the code a little bit to be shorter and more readable.